In [12]:
from scipy.optimize import minimize

fun = lambda x: (x[0] - x[1])**2 + (x[1] +x[2]-2)**2 + (x[3]-1)**2 + (x[4]-1)**2

##constrains
cons = ({'type': 'eq', 'fun': lambda x:   x[0] + 3 * x[1] },
        {'type': 'eq', 'fun': lambda x:   x[2] + x[3] - 2 * x[4] },
        {'type': 'eq', 'fun': lambda x:   x[1] -  x[4] })
##bounds
bnds = ((-10, 10), (-10, 10),(-10, 10),(-10, 10),(-10, 10))

##initial guess(2,2,2,2,2)
res = minimize(fun, (2, 2, 2, 2, 2), method='SLSQP', bounds=bnds, constraints=cons)
res

     fun: 4.09302328745791
     jac: array([-2.04625618, -0.18625832, -2.2325145 , -2.23279339, -1.48843598])
 message: 'Optimization terminated successfully'
    nfev: 43
     nit: 7
    njev: 7
  status: 0
 success: True
       x: array([-0.76734607,  0.25578202,  0.62796073, -0.11639669,  0.25578202])

The function $$fun= (x_1-x_2)^2 + (x_2+x_3-2)^2 + (x_4-1)^2 + (x_5-1)^2$$  

reaches to its min value $$min\quad fun=4.093$$ ,

when 
$$
 \begin{bmatrix}
   x_1 \\\\
   x_2 \\\\
   x_3 \\\\
   x_4 \\\\
   x_5 
  \end{bmatrix} \quad=\quad\begin{bmatrix}
   -0.767 \\\\
   0.256 \\\\
   0.628 \\\\ 
   -0.116 \\\\
   0.256
  \end{bmatrix}$$
